### PR TITLE
executor: add diagnosis rule to detect cluster critical errors

### PR DESCRIPTION
**Pull request by lonng**
### What problem does this PR solve? 

This PR adds a new diagnosis rule, which is used to detect whether critical errors occurred in the cluster. We will detect the following metrics tables in the current implementation:

- `tidb_failed_query_opm`
- `tikv_critical_error`
- `tidb_panic_count`
- `tidb_binlog_error_count`
- `pd_cmd_fail_ops`
- `tidb_kv_region_error_ops`
- `tidb_lock_resolver_ops`
- `tikv_scheduler_is_busy`
- `tikv_coprocessor_is_busy`
- `tikv_channel_full_total`
- `tikv_coprocessor_request_error`
- `tidb_schema_lease_error_opm`
- `tidb_transaction_retry_error_ops`
- `tikv_grpc_errors`

### What is changed and how it works?

Check the metrics table and check whether some errors occurred in the past.

### Check List 

Tests 

 - Unit test

Release note

 - Add diagnosis rule `critical-error` which is used to detect cluster critical errors

### CODE REVIEW

File: infoschema/metric_table_def.go [hunk 1]
```diff
@@ -1384,11 +1384,6 @@ var MetricTableMap = map[string]MetricTableDef{
 		PromQL: `sum(rate(tikv_coprocessor_request_duration_seconds_count{$LABEL_CONDITIONS}[$RANGE_DURATION])) by (req,instance)`,
 		Labels: []string{"instance", "req"},
 	},
-
-	"tikv_cop_total_request_errors": {
```
Comment (crazycs520) [inline infoschema/metric_table_def.go]: Why remove this? It is different from `tikv_coprocessor_is_busy`.

Comment (lonng) [reply infoschema/metric_table_def.go]: They have the same expression.

Comment (lonng) [reply infoschema/metric_table_def.go]: reverted

File: executor/cluster_reader_test.go [hunk 2]
```diff
@@ -76,25 +75,46 @@ func (s *testClusterReaderSuite) TestMetricTableData(c *C) {
...
+		},
+		{
+			sql: "select time,instance,quantile,value from tidb_query_duration where quantile=0.5",
+			exp: []string{
+				"2019-12-23 20:11:35.000000 127.0.0.1:10080 0.5 0.1",
+			},
+		},
+	}
+
+	for _, cas := range cases {
```
Comment (Deardrops) [inline executor/cluster_reader_test.go]:
```suggestion
	for _, case := range cases {
```

Comment (lonng) [reply executor/cluster_reader_test.go]: `case` is a keyword.

**Pull request merged**